### DISCUSSION TITLE
Usage with Clean architecture

### DISCUSSION THREAD

**Discussion by abd3lraouf** [orbit-mvi/orbit-mvi#34] [Q&A] [answered]
Hello again,

Thanks for the great library and your continuous support, Will be using it definitely from now on.

I'm using clean architecture in my application which is divided into `domain`, `presentation`, `ui` layers.
Yet I've some questions:

1. Isn't the usage of `parcelize` and `:Parcelable` in the domain layer adds another dependency on the android platform, which promotes the module to an `android-library` rather than a `jvm-library`?

**Should I use a mapper here ?**

2. Lets say I've a usecase class, which I want to talk to from the `ViewModel`. From which point should I call the useCase's `execute` method? that should be from the `intent` extension?

THat's it.

**Comment by abd3lraouf** [comment]
Just found a **variation** of the `container` that takes `Any`, I think `Question 1.` is resolved.

**Comment by Rosomack** [comment, marked as answer]
Hey, thanks for your questions:

Re 1) 

Overall it depends on what you want to achieve and how convenient you want it to be. In our sample apps we went for simplicity and convenience.

If you want to be strict about Clean Architecture I'd avoid using `Parcelable`s in the domain layer. That comes with a drawback though, because you'd need to write parcelization code yourself somehow.

An alternative is using `Serializable` which should be portable across JVM and Android at the cost of a little efficiency.

Re 2)

Yes, exactly.  `intent`'s block is a suspending lambda, so you can call your use cases from there.

**Reply by abd3lraouf** [reply]
Fair enough. I'll adopt this library from now on and see where it takes me.

Thanks @Rosomack
